src/protectAPI.coffee

```diff
@@ -35,7 +35,7 @@ reFlags = /\w*$/
 
 # Hacky reimplementation of lodash's cloneDeep, minus the parts we don't need, plus limiting clones to apiProperties.
 module.exports.createAPIClone = createAPIClone = (aether, value) ->
-  return value unless _.isObject value
+  return value if not _.isObject(value) or 'next' of value
```

Comment:
Pretty hacky way to avoid breaking the iterators that are returned when you call a spec.-compliant generator function. More specifically, these iterator objects inherit from a custom prototype, so we can't clone them or they would lose the `next` method.

I believe there has been a lot of discussion about custom prototypes and API protection, and as far as I can see, it is not possible to out-right clone an object and preserve all of its behavior. It is indeed possible to create a new object with all the properties and the prototype chain of another given object, but that given object's constructor may have some custom initialization logic and side-effects that can't be reproduced in this way. There are, obviously, several other complications.

To avoid this long and unfruitful discussion, I've just added a quick and dirty hack for now.